### PR TITLE
`Tutorial groups`: Fix a usability issue for creating new tutorial for campus and language suggestions

### PR DESCRIPTION
**Pull request by max-bergmann**
#### General

- [X] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [X] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://ls1intum.github.io/Artemis/dev/guidelines/language-guidelines/).
- [X] I chose a title conforming to the [naming conventions for pull requests](https://ls1intum.github.io/Artemis/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [X] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [X] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/server/).
- [X] I documented the Java code using JavaDoc style.
#### Client
- [X] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [X] I followed the [coding and design guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client/).
- [X] Following the [theming guidelines](https://ls1intum.github.io/Artemis/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [X] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [ ] I translated all newly inserted strings into English and German.

### Motivation and Context

This PR is designed to improve usability when creating tutor groups. The PR offers the default selection of English and German. For the campus search, the default values have been extended. I.e. all possible values of the campus the instructor is involved in are now offered - not only the possibilities of the respective course

### Description

fixes https://github.com/ls1intum/Artemis/issues/7292

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Courses where the user is instructor
- 2 Tutor Groups

1. Log in to Artemis
2. Navigate to Course Administration
3. Create one Tutorial --> check whether the default values for english and german are available.  
4. Switch to 2nd course --> create another Tutorial 
5. Check if the previouly inserted campus is visible. 
6. Repeat steps for editing of a tutorial

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| create-tutorial-group.component.ts | 98% | ✅              |

### Screenshots
image
image

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java [hunk 1]
```diff
@@ -161,7 +162,7 @@ public ResponseEntity<Set<String>> getUniqueLanguageValues(@PathVariable Long co
     public ResponseEntity<List<TutorialGroup>> getAllForCourse(@PathVariable Long courseId) {
         log.debug("REST request to get all tutorial groups of course with id: {}", courseId);
         var course = courseRepository.findByIdElseThrow(courseId);
-        var user = userRepository.getUserWithGroupsAndAuthorities();
+        var user = userRepository.getUser();
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupResource.java:165]: The groups are needed in the next call (checkHasAtLeastRoleInCourseElseThrow), which currently leads to an additional database request.

**Review by sarpsahinalp — COMMENTED** [review]
I can't see the changes on ts3
image

Default values are not selectable
image

File: src/main/webapp/app/course/tutorial-groups/tutorial-groups-management/tutorial-groups/crud/tutorial-group-form/tutorial-group-form.component.ts [hunk 2]
```diff
@@ -367,6 +367,9 @@ export class TutorialGroupFormComponent implements OnInit, OnChanges, OnDestroy
             ),
         ).subscribe((languages: string[]) => {
             this.languages = languages;
+            // default values for English & German
+            if (!languages.includes('English')) this.languages.push('English');
+            if (!languages.includes('German')) this.languages.push('German');
```
Comment (Strohgelaender) [inline src/main/webapp/app/course/tutorial-groups/tutorial-groups-management/tutorial-groups/crud/tutorial-group-form/tutorial-group-form.component.ts:372]:
```suggestion
            if (!languages.includes('English')) {
                this.languages.push('English');
            }
            if (!languages.includes('German')) {
                this.languages.push('German');
            }
```

**Review by JanaNF — CHANGES_REQUESTED** [review]
Maunal tested on legacy ts2. Worked mostly as expected.

But for the languages: If I added a new language (e.g. Russisch) it did not show up as a default value for the other tutorial (like it did with the locations). I don't know if this is something you want to happen or not. Please let me know.

**Comment by max-bergmann** [comment]
> Maunal tested on legacy ts2. Worked mostly as expected.
> 
> But for the languages: If I added a new language (e.g. Russisch) it did not show up as a default value for the other tutorial (like it did with the locations). I don't know if this is something you want to happen or not. Please let me know.

Thanks for your review. Originally, the change was only meant for several campuses in different courses, but I have now included the change for the languages as well.

**Review by vinceclifford — CHANGES_REQUESTED** [review]
I reviewed the PR previously with Jana when the selection between German and English was still working. 

I tested on TS3 and the requirements (the language selection) were not fulfilled. 
Firstly, I can't choose between English and German. There is no drop-down menu. Screenshot of me entering German down below. 
Secondly, previous inputs are not "transferred" to different courses. If I select Straubing as the campus in Course1 and then go to Course2, I will not get Straubing recommended. However, if I create another tutorial group in Course1 I will see my previous input. I don't know if this is as it should be. 
However, the selection of German and English is not shown to me. 

Screenshot 2023-10-29 at 18 28 23

**Review by JanaNF — APPROVED** [review]
Tested again on ts2. Worked as expected

**Review by milljoniaer — DISMISSED** [review]
Tested ts4, works as expected

**Review by Strohgelaender — CHANGES_REQUESTED** [review]
Please implement my change request above, this currently leads to unneccesary database requests.

**Review by rstief — APPROVED** [review]
Tested on legacy ts3 in testing session.

**Review by MaximilianAnzinger — APPROVED** [review]
Tested during testing session

**Review by lennart-keller — APPROVED** [review]
Tested in testing session

**Pull request merged**